tests/tests_unit/test_api/test_assets.py

```diff
@@ -109,6 +109,43 @@ def test_list(self, mock_assets_response):
         assert "bla" == jsgz_load(mock_assets_response.calls[0].request.body)["filter"]["name"]
         assert mock_assets_response.calls[0].response.json()["items"] == res.dump(camel_case=True)
 
+    def test_list_root(self, mock_assets_response):
+        res = ASSETS_API.list(root_ids=[{"id": 1}, {"externalId": "abc"}], limit=10)
+        calls = mock_assets_response.calls
+        assert 1 == len(calls)
+        assert {"cursor": None, "limit": 10, "filter": {"rootIds": [{"id": 1}, {"externalId": "abc"}]}} == jsgz_load(
+            calls[0].request.body
+        )
+
+    def test_call_root(self, mock_assets_response):
+        for a in ASSETS_API(root_ids=[{"id": 1}, {"externalId": "abc"}], limit=10):
```

Comment:
ASSETS_API.__call__(root_ids=[{"id": 1}, {"externalId": "abc"}], limit=10)

Reply:
you mean an explicit __call__ ? don't think it matters, but ok

Reply:
no, sorry. just meant that iterating over it is redundant.

Reply:
it doesn't seem to execute/call when you don't iterate over anything. iterator magic i suppose

Reply:
changed it to list() which works too